SP-8 Overhaul

KillerTheFlareon opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
<!-- What do you propose to change with your PR? -->
The sprites for the SP-8 were low-quality, with a million color variations. This changes the sprite into something more resembling a SS14 firearm, as well as the magazine. I also added craftable ammo/box/magazines for the gun.
## Why we need to add this
<!-- What is the reason for adding these changes? Please post links to Discussions as well as Bug Reports here. Please describe how this will change the game balance. -->
Warden almost NEVER uses his sidearm due to the extremely limited supply of ammo (no way to make more) so I sat down and sprited some magazines and boxes that are now craftable for the gun in the secfab. The gun is now actually usable, and doesn't look extremely out of style.

I also renamed the SP-8 into "Due Process" because SP-8 did not fit the naming scheme of guns in ss14.
## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->
![yfilyiu;](https://github.com/user-attachments/assets/748c5743-5da1-4cba-b4b4-898092b02fef)
## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
<!--
If you want the players to know about changes made in this PR, specify them using the template outside the comment. Short and informative.
-->
:cl: Killer Tamashi
- add: Added .40 Boxes to secfab.
- add: Added .40 Magazines to secfab.
- tweak: Changed SP-8's appearance to a pistol more in style.
- tweak: Renamed SP-8 to "Due Process" to be more in line with SS14 naming scheme.

KillerTheFlareon committed:
Overhauls SP-8 into Due Process

Also makes the ammo/magazines craftable in secfab so the warden can actually USE the damn thing.

KillerTheFlareon committed:
forgot to change meta

yeah, i'm tired.

Rinary1 commented on Resources/Prototypes/Roles/Jobs/Security/warden.yml:
```diff
@@ -34,9 +34,9 @@
     id: WardenPDA
     ears: ClothingHeadsetSecurity
-    pocket1: WeaponPistolSP8
+    pocket1: WeaponPistolDP
```
why you recalled it?

Rinary1 commented on Resources/Prototypes/Roles/Jobs/Security/warden.yml:
```diff
@@ -34,9 +34,9 @@
     - Flash
-    - MagazinePistol40Nr
-    - MagazinePistol40Nr
+    - MagazinePistol40SP
+    - MagazinePistol40SP
```
why you recalled ammo?

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Objects/Weapons/Guns/Ammunition/Boxes/pistol.yml:
```diff
@@ -14,14 +14,89 @@
-  parent: BaseMagazineBoxPistol
-  id: MagazineBoxPistol40Nr
-  name: ammunition box (40 NR)
+  parent: BaseItem
+  id: BaseMagazineBoxPistol40
+  name: ammunition box (.40)
```
why you recalled it

Rinary1 reviewed (changes requested):
Why you recalled magazine name/name of pistol, also sp-8 have a normal sprite in ss14 style(it taked from paradise station), also with ammo magazine, your variant don't have any colors of old variation, also magazine looks worse

KillerTheFlareon commented:
> Why you recalled magazine name/name of pistol, also sp-8 have a normal sprite in ss14 style(it taked from paradise station), also with ammo magazine, your variant don't have any colors of old variation, also magazine looks worse

I have answered this in #dev-chat but for clarity I will also say it here;

The magazine looked horrible, I don't know what you mean by it "looked good" it was massive, barely shaded and looked like a chunk of plastic (which you can see in #dev-chat for comparison against normal ss14 pistol magazines)
My new magazine has the exact same colors as the old magazine, actually. You can once again see the comparison in #dev-chat

The sprite was remade because, again, it looked really bad. The pistol was blocky and barely looked like a gun.
As for the colors, I can't make "in-hand" sprites with 11 different color variations, and to be honest they all didn't look that good with the color choice anyway.

As for the renaming, that was done because it didn't even follow basic naming conventions for ammo types (.40 SP) instead of (40 NR). The name itself was changed because SP-8 looks odd among "Drozd, Lector, Krammer, Enforcer, ect.."
All in all it looked really out of place. Just because it was taken from paradise doesn't mean it looked good.

KillerTheFlareon committed:
Made mags thinner.

People correctly pointed out they were too big to fit in the gun, so I fixed that.

KillerTheFlareon committed:
Merge branch 'Starlight' into sp-8-overhaul

KillerTheFlareon committed:
Revert "Merge branch 'Starlight' into sp-8-overhaul"

This reverts commit ba2592fa81905e58e66ec8497bf6f64d35d12906, reversing
changes made to 6230903883e3a1de1ecf1e778df72a93766a5b96.

KillerTheFlareon commented:
I accidentally blew up this pr with a bad merge. Don't think I can fix it so I might just make a new pr.

KillerTheFlareon commented:
https://github.com/ss14Starlight/space-station-14/compare/Starlight...KillerTheFlareon:KillerT-SS14-Starlight:sp-8-overhaul-2?expand=1
Bricked it entirely by accidentally pulling unstable breanch. New PR is here:
